Assets/Scripts/Card/SimpleCard.cs

```diff
@@ -2,27 +2,73 @@
 using System.Collections;
 using System.Collections.Generic;
 using UnityEngine;
-
 public class SimpleCard : MonoBehaviour
 {
     GameObject cardFace;
-    Action<SimpleCard> clickAction;
     public Guid cardGuid;
+    public bool isSelected { get; private set; } = false;
+    public bool isPlacedOnBattlefield { get; private set; } = false;
 
-    public void Setup(GameObject _cardFaceTemplate, Action<SimpleCard> _clickAction, Guid _cardGuid)
+    public void Setup(GameObject _cardFaceTemplate, Guid _cardGuid)
     {
-        cardFace = Instantiate(_cardFaceTemplate, transform);
-        clickAction = _clickAction;
+        cardFace = Instantiate(_cardFaceTemplate, this.transform);
         cardGuid = _cardGuid;
     }
 
-    public void UpdateClickAction(Action<SimpleCard> _clickAction)
+    private void OnMouseDown()
     {
-        clickAction = _clickAction;
+        this.isSelected = true;
     }
 
-    private void OnMouseDown()
+    private void OnMouseUp()
     {
-        clickAction(this);
+        Debug.Log("Mouse button is released; Do not move card " + this.cardGuid);
+
+        this.isSelected = false;
+        IInteractable? collisionObject = this.GetColissionObject();
+
+        if (collisionObject == null)
+            this.PutCardToHand();
+        else
+            collisionObject.RecieveObject(this);
+    }
+
+    private void Update()
+    {
+        if (this.isSelected)
+        {
+            Debug.Log("Mouse button is clicked; Updating card position " + this.cardGuid);
+
+            Vector3 coursorePosition = Camera.main.ScreenToWorldPoint(Input.mousePosition);
+            this.transform.position = new Vector3(coursorePosition.x, coursorePosition.y, -5.0f);
+
+            Debug.Log("Mouse position " + Input.mousePosition + " Card position " + this.transform.position);
+        }
+    }
+
+    private IInteractable? GetColissionObject()
+    {
+        RaycastHit hit;
+        bool isHit = Physics.Raycast(
+            this.transform.position,
+            transform.TransformDirection(Vector3.forward),
+            out hit,
+            Mathf.Infinity,
+            Physics.DefaultRaycastLayers
+        );
+
+        Debug.DrawRay(this.transform.position, transform.TransformDirection(Vector3.forward) * 20.0f, isHit ? Color.yellow : Color.red);
+
+        if (hit.collider == null)
```

Comment:
According to [documentation](https://docs.unity3d.com/ScriptReference/Physics.Raycast.html), "If true is returned, hitInfo will contain more information about where the closest collider was hit". Are you sure it is safe to keep it like this and do not check for `isHit` equals true?

Reply:
Tbh I agree, prolly check `isHit` variable would be enough.